diff --git a/src/input/actions.rs b/src/input/actions.rs
index 3d017697..035ddd32 100644
--- a/src/input/actions.rs
+++ b/src/input/actions.rs
@@ -204,7 +204,7 @@ impl State {
             }
 
             Action::NextWorkspace => {
-                if let Some(direction) = pattern.inferred_direction()
+                if let Some(direction) = direction
                     && (((direction == Direction::Left || direction == Direction::Right)
                         && self.common.config.cosmic_conf.workspaces.workspace_layout
                             == WorkspaceLayout::Vertical)
@@ -228,7 +228,7 @@ impl State {
                 );
                 if next.is_err()
                     && propagate
-                    && let Some(inferred) = pattern.inferred_direction()
+                    && let Some(inferred) = direction
                 {
                     self.handle_shortcut_action(
                         Action::SwitchOutput(inferred),
@@ -243,7 +243,7 @@ impl State {
             }
 
             Action::PreviousWorkspace => {
-                if let Some(direction) = pattern.inferred_direction()
+                if let Some(direction) = direction
                     && (((direction == Direction::Left || direction == Direction::Right)
                         && self.common.config.cosmic_conf.workspaces.workspace_layout
                             == WorkspaceLayout::Vertical)
@@ -267,7 +267,7 @@ impl State {
                 );
                 if previous.is_err()
                     && propagate
-                    && let Some(inferred) = pattern.inferred_direction()
+                    && let Some(inferred) = direction
                 {
                     self.handle_shortcut_action(
                         Action::SwitchOutput(inferred),
@@ -338,7 +338,7 @@ impl State {
             }
 
             x @ Action::MoveToNextWorkspace | x @ Action::SendToNextWorkspace => {
-                if let Some(direction) = pattern.inferred_direction()
+                if let Some(direction) = direction
                     && (((direction == Direction::Left || direction == Direction::Right)
                         && self.common.config.cosmic_conf.workspaces.workspace_layout
                             == WorkspaceLayout::Vertical)
@@ -385,7 +385,7 @@ impl State {
                     }
                     Ok(None) => {}
                     Err(_) if propagate => {
-                        if let Some(inferred) = pattern.inferred_direction() {
+                        if let Some(inferred) = direction {
                             self.handle_shortcut_action(
                                 if matches!(x, Action::MoveToNextWorkspace) {
                                     Action::MoveToOutput(inferred)
@@ -430,7 +430,7 @@ impl State {
             }
 
             x @ Action::MoveToPreviousWorkspace | x @ Action::SendToPreviousWorkspace => {
-                if let Some(direction) = pattern.inferred_direction()
+                if let Some(direction) = direction
                     && (((direction == Direction::Left || direction == Direction::Right)
                         && self.common.config.cosmic_conf.workspaces.workspace_layout
                             == WorkspaceLayout::Vertical)
@@ -476,7 +476,7 @@ impl State {
                     }
                     Ok(None) => {}
                     Err(_) if propagate => {
-                        if let Some(inferred) = pattern.inferred_direction() {
+                        if let Some(inferred) = direction {
                             self.handle_shortcut_action(
                                 if matches!(x, Action::MoveToPreviousWorkspace) {
                                     Action::MoveToOutput(inferred)
